src/rviz/TrajectoryMarker.cpp

```diff
@@ -0,0 +1,259 @@
+#include "aikido/rviz/TrajectoryMarker.hpp"
+
+#include "aikido/rviz/shape_conversions.hpp"
+#include "aikido/statespace/dart/MetaSkeletonStateSpace.hpp"
+
+namespace aikido {
+namespace rviz {
+
+//==============================================================================
+TrajectoryMarker::TrajectoryMarker(
+    interactive_markers::InteractiveMarkerServer* markerServer,
+    const std::string& frameId,
+    trajectory::ConstTrajectoryPtr trajectory,
+    const dart::dynamics::MetaSkeleton& targetSkeleton,
+    const dart::dynamics::Frame& frame,
+    const Eigen::Vector4d& rgba,
+    double thickness,
+    std::size_t numLineSegments)
+  : mMarkerServer(markerServer)
+  , mInteractiveMarker()
+  , mFrameId(frameId)
+  , mTrajectory(nullptr)
+  , mNumLineSegments()
+  , mSkeleton(targetSkeleton)
+  , mFrame(frame)
+  , mNeedUpdate(true)
+  , mNeedPointsUpdate(true)
+{
+  using visualization_msgs::InteractiveMarkerControl;
+  using visualization_msgs::Marker;
+
+  // TODO(JS): Check the compatibility of statespaces of the skeleton and the
+  // spline
+
+  // Setting invariant properties
+  mInteractiveMarker.header.frame_id = mFrameId;
+  mInteractiveMarker.name = "Frame[Trajectory]";
+  // TODO: Assign trajectory name once available
+  mInteractiveMarker.pose.orientation.w = 1;
+  mInteractiveMarker.scale = 1;
+
+  mInteractiveMarker.controls.resize(1);
+  InteractiveMarkerControl& control = mInteractiveMarker.controls.front();
+  control.orientation.w = 1;
+  control.orientation_mode = InteractiveMarkerControl::INHERIT;
+  control.interaction_mode = InteractiveMarkerControl::NONE;
+  control.always_visible = true;
+  control.markers.resize(1);
+
+  auto& marker = getMarker();
+  marker.type = Marker::LINE_STRIP;
+  marker.pose.orientation.w = 1.0;
+
+  // Setting variant properties
+  setTrajectory(std::move(trajectory));
+  setRGBA(rgba);
+  setThickness(thickness);
+  setNumLineSegments(numLineSegments);
+}
+
+//==============================================================================
+TrajectoryMarker::~TrajectoryMarker()
+{
+  mMarkerServer->erase(mInteractiveMarker.name);
+}
+
+//==============================================================================
+void TrajectoryMarker::setTrajectory(trajectory::ConstTrajectoryPtr trajectory)
+{
+  using visualization_msgs::Marker;
+
+  if (trajectory)
+  {
+    auto statespace = trajectory->getStateSpace();
+    if (!std::dynamic_pointer_cast<statespace::dart::MetaSkeletonStateSpace>(
```

Comment:
Maybe we can add a method in `MetaSkeletonStateSpace` to actually check whether a `MetaSkeleton` is compatible. Then we can use that check as well.

Reply:
Sounds good to me. As the check is not available yet, I'm just checking the dimensions and adding a comment suggesting using more comprehensive check once available.

Reply:
https://github.com/personalrobotics/aikido/issues/292